Feature/resumes

slowtree999 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **신기능**
  - 이력서 메타데이터 조회 API 추가(기술스택·이력서/경력/학위/숙련도/프로젝트유형 등).
  - 이력서 저장·수정·상세(상세페이로드)·목록(페이징·썸네일)·복원(관리자) RESTful 엔드포인트 제공.

- **변경(리팩터링)**
  - 이력서 도메인·응답 스키마 전면 개편(상세/썸네일/페이지 도입, 상태 기반 흐름).

- **기능 제거**
  - AI 기반 이력서 제안 및 파일 업로드(임포트) 관련 API/클라이언트 제거.

- **기타**
  - 이력서·Lambda 관련 예외 처리 핸들러 추가(사용자용 에러 응답 개선).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

slowtree999 committed:
[AICH-29] 람다 추가

msw-Hub committed:
[AICH-29] fix :  람다 코드 수정

- 람다 연결 부분만 수정하였습니다.
- local.yml도 수정본으로 바꾸셔야합니다

slowtree999 committed:
[AICH-29] fix :  실행 오류 수정

slowtree999 committed:
[AICH-34] 의존성 분리 예외 처리 개선

slowtree999 committed:
[AICH-34] 이력서 도메인 개선 테스트 코드 추가 프롬프트 수정중

slowtree999 committed:
[AICH-34] 이력서 dto 중복 수정 도메인 구조 개선

slowtree999 committed:
[AICH-34] 코드 최적화 버그 수정 메서드 분리

msw-Hub committed:
[AICH-36] fix : 이력서 전면 수정 (엔티티부터~기능 구현까지)

현재는 저장 로직만 구현

msw-Hub committed:
[AICH-66] feat : 이력서 수정(덮어쓰기) 구현 및 쿼리dsl 사용으로 인한 빌드그리들 수정

msw-Hub committed:
[AICH-65] feat : 이력서 소프트 삭제 구현

msw-Hub committed:
[AICH-36] fix : 이력서 request로 받을때, 벨리데이션 체크를 서비스단에서 하도록 수정

msw-Hub committed:
[AICH-83] feat : 작성한 이력서 리스트를 불러오기 (페이징) 구현

msw-Hub committed:
[AICH-67] feat : 이력서 id로 상세조회 및 변수명 수정

msw-Hub committed:
[AICH-84] feat : 이력서 만들때 메타데이터 조회

msw-Hub committed:
[AICH-84] fix : 학력의 성적 타입이 Double이라 발생한 문제 수정

slowtree999 committed:
[AICH-38] ai 이력서 제안

slowtree999 committed:
[AICH-38] 이력서 ai 추천 키워드 오류 수정

slowtree999 committed:
[AICH-38] 이력서 ai 제안 프롬프트 수정

msw-Hub committed:
[AICH-85] feat : 이력서 소프트삭제 복구 기능 및 스케쥴링으로 하드삭제 (주석처리)

추가적으로 이전에 있던 테스트 코드 삭제했습니다

slowtree999 committed:
[AICH-38] 이력서 ai 추천 테스트 코드

msw-Hub committed:
Merge branch 'develop' into feature/resumes

slowtree999 committed:
[AICH-38] 이력서 ai 제안

slowtree999 committed:
[AICH-38] 이력서 ai 제안 삭제

msw-Hub commented on CVMento/src/main/java/com/cvmento/global/aws/LambdaService.java:
```diff
@@ -0,0 +1,113 @@
+    public String invokeLambdaOcr(MultipartFile file) {
+        try {
+            // 파일을 Base64로 변환
+            String base64File = Base64.getEncoder().encodeToString(file.getBytes());
```
지금 내용으로는 파일을 base64로 변환해서 사용하는 것으로 보입니다.
그런데, 파일 크기 제한과, 파일 확장자 검증이 안보입니다.
아직 해당 서비스를 사용하는 사용처가 없기에 문제는 없어서 pr 이후에 수정하시면 될것같습니다!

slowtree999 replied:
확인했습니다 개선방안 찾아보겠습니다

msw-Hub commented on CVMento/src/main/java/com/cvmento/global/config/AwsConfig.java:
```diff
@@ -0,0 +1,23 @@
+    public AWSLambda lambdaClient() {
+        return AWSLambdaClientBuilder.standard()
+                .withRegion(region)
+                .withCredentials(InstanceProfileCredentialsProvider.getInstance())
```
해당 설정으로는 ec2에서만 동작할 것으로 보이는데, 로컬에서는 테스트를 못하는 건가요??

slowtree999 replied:
로컬환경에서는 테스트를 안해보고 일단 올려서 동작구조를 알아보고 개선하는 방향으로 생각해두었습니다 현재 이력서 부분이 대량 조정되어 로컬환경에서 테스트 환경도 구축 하는 방향으로 수정해보겠습니다

msw-Hub commented on CVMento/src/main/java/com/cvmento/global/aws/LambdaService.java:
```diff
@@ -0,0 +1,113 @@
+
+        } catch (IOException e) {
+            log.error("파일 처리 중 오류: {}", e.getMessage(), e);
+            throw new RuntimeException("파일을 읽는 중 오류가 발생했습니다.", e);
```
예외처리가 RuntimeException으로만 되어있는데, 저희 룰에 맞춰서 글로벌 예외처리로 수정하시면 될것같습니다!

slowtree999 replied:
확인했습니다 해당 부분 반영하겠습니다

slowtree999 committed:
[AICH-38] pr 코드 수정 테스트

slowtree999 committed:
[AICH-38] 코드 수정 테스트 #22

slowtree999 committed:
[AICH-38] 글로벌 예외 처리 #22

 1. LambdaException.java 생성
    - 기존 커스텀 예외 패턴을 따라 생성
    - RuntimeException을 상속받는 구조
  2. GlobalExceptionHandler.java 수정
    - LambdaException 핸들러 추가
    - HTTP 500 상태코드와 LAMBDA_SERVICE_ERROR 에러코드 반환
  3. LambdaService.java 수정
    - 모든 RuntimeException을 LambdaException으로 교체
    - import문에 LambdaException 추가
